node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/StorageMaintainer.java

```diff
@@ -227,77 +208,40 @@ long getDiskUsedInBytes(Path path) throws IOException, InterruptedException {
-                environment.pathInNodeUnderVespaHome("logs/daemontools_y"),
-                environment.pathInNodeUnderVespaHome("logs/nginx"),
-                environment.pathInNodeUnderVespaHome("logs/vespa")
+    /** Deletes old log files for vespa, nginx, logstash, etc. */
+    public void removeOldFilesFromNode(NodeAgentContext context) {
+        Path[] logPaths = {
+                context.pathInNodeUnderVespaHome("logs/elasticsearch2"),
+                context.pathInNodeUnderVespaHome("logs/logstash2"),
+                context.pathInNodeUnderVespaHome("logs/daemontools_y"),
+                context.pathInNodeUnderVespaHome("logs/nginx"),
+                context.pathInNodeUnderVespaHome("logs/vespa")
         };
 
-        for (Path pathToClean : pathsToClean) {
-            Path path = environment.pathInNodeAdminFromPathInNode(containerName, pathToClean);
-            if (Files.exists(path)) {
-                maintainerExecutor.addJob("delete-files")
-                        .withArgument("basePath", path)
-                        .withArgument("maxAgeSeconds", Duration.ofDays(3).getSeconds())
-                        .withArgument("fileNameRegex", ".*\\.log.+")
-                        .withArgument("recursive", false);
-            }
+        for (Path pathToClean : logPaths) {
+            Path path = context.pathOnHostFromPathInNode(pathToClean);
+            FileFinder.files(path)
+                    .match(olderThan(Duration.ofDays(3)).and(nameMatches(Pattern.compile(".*\\.log.+"))))
+                    .maxDepth(1)
+                    .deleteRecursively();
         }
 
-        Path qrsDir = environment.pathInNodeAdminFromPathInNode(
-                containerName, environment.pathInNodeUnderVespaHome("logs/vespa/qrs"));
-        maintainerExecutor.addJob("delete-files")
-                .withArgument("basePath", qrsDir)
-                .withArgument("maxAgeSeconds", Duration.ofDays(3).getSeconds())
-                .withArgument("recursive", false);
-
-        Path logArchiveDir = environment.pathInNodeAdminFromPathInNode(
-                containerName, environment.pathInNodeUnderVespaHome("logs/vespa/logarchive"));
-        maintainerExecutor.addJob("delete-files")
-                .withArgument("basePath", logArchiveDir)
-                .withArgument("maxAgeSeconds", Duration.ofDays(31).getSeconds())
-                .withArgument("recursive", false);
-
-        Path fileDistrDir = environment.pathInNodeAdminFromPathInNode(
-                containerName, environment.pathInNodeUnderVespaHome("var/db/vespa/filedistribution"));
-        maintainerExecutor.addJob("delete-files")
-                .withArgument("basePath", fileDistrDir)
-                .withArgument("maxAgeSeconds", Duration.ofDays(31).getSeconds())
-                .withArgument("recursive", true);
-    }
+        FileFinder.files(context.pathOnHostFromPathInNode(context.pathInNodeUnderVespaHome("logs/vespa/qrs")))
```

Comment:
Consider making a method pathOnHostFromPathInNodeUnderVespaHome()

Reply:
Will do. Might get a little polluted to also have that method accept either `String` or `Path`